tools/slice_image/slice_image.xml

```diff
@@ -0,0 +1,89 @@
+<tool id="slice_image" name="Slice Image" version="0.1"> 
+  <description>Slice image</description>
+  <requirements> 
+    <requirement type="package" version="0.14.0">scikit-image</requirement>
+    <requirement type="package" version="1.15.4">numpy</requirement>
+    <requirement type="package" version="1.1.0">scipy</requirement>
+  </requirements>
+  <command> 
+        <![CDATA[
+        mkdir ./out_dir &&
+
+        python "$__tool_directory__/slice_image.py" 
+        '$input_file' ./out_dir
+        --stride $stride --window_size $patch_size --bg_thresh $bg_thresh 
+        --limit_slices $limit_slices.bool 
+        #if $limit_slices.bool == "true":
+          --n_thresh $limit_slices.n_tresh
+        #end if
+        #if $control_rng.bool == "true":
+          --seed $control_rng.seed
+        #end if
+        ]]> 
+  </command>
+  <inputs> <!--inputs as seen in the interface-->
+    <param name="input_file" type="data" format="tiff,png,jpg,bmp" label="Input image"/> 
+    <!--<param name="out_folder" type="text" label="Output folder for image slices"/> 
+    <param name="label_file" type="data" format="tiff,png,jpg,bmp" label="Auxiliary label file to split in the same way"/>--> 
+    <param name="stride" type="integer" optional="true" value="16"/>
+    <param name="patch_size" type="integer" optional="true" value="64" label="Size of resulting (quadratic-shaped) slices"/>
+    <param name="bg_thresh" type="float" optional="true" value="0" label="Threshold to not be considered background"/>
+    <conditional name="limit_slices">
+          <param name="bool" type="boolean" checked="false" truevalue="true" label="Should the number of slices be limited?"/>
+          <when value="true">
+            <param name="n_thresh" type="integer" value="500" label="Maximum amount of slices"/>
```

Comment:
On the other side, if you really just have this one parameter, why not making this parameter `optional=True` and remove the entire conditional?

Reply:
I though it was good practice to only offer an option (i.e. n_thresh) when it also has an effect. But it seems more convenient your way, so I will change it

Reply:
Not sure we have a guidline for this. My reasoning is that if you need this option you now need to click twice. If you want to hide parameters, just visually, there is also the `section` approach.